src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -80,6 +80,22 @@ get_hw_gen(const pt::ptree& aie_meta)
   return aie_meta.get<uint8_t>("aie_metadata.driver_config.hw_gen");
 }
 
+uint32_t
+get_partition_id(const pt::ptree& aie_meta)
+{
+  uint8_t num_col = aie_meta.get<uint8_t>("aie_metadata.driver_config.partition_num_cols");
+  uint8_t start_col = 0; 
+  auto overlay_start_cols = aie_meta.get_child_optional("aie_metadata.driver_config.partition_overlay_start_cols");
+  if (overlay_start_cols) {
+    for (const auto& col : *overlay_start_cols) {
+      start_col = col.second.get_value<uint8_t>();
+      break; // Assuming we'll have only one value
+    }
```

Comment:
Why a for loop given the assumption?

Reply:
I added for loop considering that more entries might get added in future for this field. But for now, since we have only one value, I'll remove the for loop